docs: mem module table lists three functions that do not exist (free, make, size_of)

## Problem

The mem module table in `STANDARD.md` (section 9.24) lists three functions the compiler does not have. Calling `mem.free`, `mem.make`, or `mem.size_of` fails with:

```
error[E4005]: module 'mem' has no function named 'free'
```

None of the three should be added to the compiler:

- **`free`** — documented as "alias for destroy". Grayscale has no stdlib function aliases.
- **`make`** — same signature and description as the existing `init`.
- **`size_of`** — already a global builtin, correctly documented at `STANDARD.md:2839`. The mem module entry is a duplicate in the wrong table.

The fix is to remove the rows.

## What to change

In `STANDARD.md`, delete lines 3806-3808:

```
| `make` | `(arena Arena, Type) -> ^Type` | Allocate zero-initialized value of Type in arena |
| `free` | `(arena Arena)` | Free an arena (alias for destroy) |
| `size_of` | `(Type) -> int` | Size of type in bytes |
```

Leave the rest of the table alone. `size_of` stays where it is at line 2839.

Documentation only — no source files change.

--- a/STANDARD.md
+++ b/STANDARD.md
@@ -3737,9 +3737,6 @@ Arena-based memory allocation. Compiler-only feature.
 | `usage` | `(arena Arena) -> int` | Return the number of bytes currently used |
 | `init` | `(arena Arena, Type) -> ^Type` | Allocate a zero-initialized value of `Type` in the arena |
 | `alloc` | `(arena Arena, value T) -> ^T` | Allocate a copy of value in the arena |
-| `make` | `(arena Arena, Type) -> ^Type` | Allocate zero-initialized value of Type in arena |
-| `free` | `(arena Arena)` | Free an arena (alias for destroy) |
-| `size_of` | `(Type) -> int` | Size of type in bytes |
 | `raw_copy` | `(dest ptr, src ptr, n int)` | Copy `n` bytes from `src` to `dest` |
 | `zero` | `(p ptr, n int)` | Zero out `n` bytes at `p` |
 | `fill` | `(p ptr, value int, n int)` | Set `n` bytes at `p` to `value` |

